app/src/plotter.py

```diff
@@ -257,52 +271,91 @@ def fit_data(self):
+                axis.axis_label_text_font_style = self.config["text"]["font_style"]
+                axis.major_label_text_font = self.config["text"]["font"]
+                axis.major_label_text_font_style = self.config["text"]["font_style"]
+                axis.axis_line_width = self.config["axis"]["axis_line_width"]
+                axis.axis_line_color = self.config["background_fill_color"]
 
             # Add lines to the plot
-            colors = Category20[20]
+            colors = overwrite_colors
             for i, line_name in enumerate(self.config["filters"]["multi_choice"]):
-                self.plots[code].line(x="x", y=line_name, source=self.fitted_data[code], color=colors[i], legend_label=line_name)
+                self.plots[code].line(
+                    x="x", y=line_name, source=self.fitted_data[code], color=colors[i], line_width=4)
+                self.plots[code].circle(
+                    x="x", y=line_name, source=self.fitted_data[code], color=colors[i], size=16)
+                self.plots[code].circle(
+                    x="x", y=line_name, source=self.fitted_data[code], color=self.config["background_fill_color"], size=12)
 
     def create_filters(self):
-        self.filters["single_choice"] = panel.widgets.RadioBoxGroup(
+        filters_single_choice = panel.widgets.RadioBoxGroup(
             name="Select unit", options=self.config["filters"]["single_choice"]
         )
 
-        # Create multi choice filters
-        self.filters["multi_choice"] = panel.widgets.CheckBoxGroup(
-            name="Select branches",
-            options=self.config["filters"]["multi_choice"],
-            value=self.config["filters"]["multi_choice"]
-        )
+        filters_multi_choice = panel.Column(*[
+            panel.Row(
+                panel.pane.HTML(
+                    '<div class="legend-field" style="background-color:{};"></div>'.format(color)),
+                panel.widgets.Checkbox(name=option, value=True,
+                                       css_classes=["legend-checkbox"])
+            )
+            for color, option in zip(overwrite_colors, self.config["filters"]["multi_choice"])
+        ])
+
+        self.filters["multi_choice"] = Accordion(
+            ("Branchen auswählen", filters_multi_choice), header_color="#1E3791", active_header_background="#F6F6F6", header_background="#F6F6F6", active=[0])
+        self.filters["single_choice"] = Accordion(
+            ("Einheiten auswählen", filters_single_choice), header_color="#1E3791", active_header_background="#F6F6F6", header_background="#F6F6F6")
 
         # Add interactivity
-        self.filters["multi_choice"].param.watch(self.update_filters, "value")
-        self.filters["single_choice"].param.watch(self.update_filters, "value")
-        self.filters["single_choice"].param.watch(self.update_y_range, "value")
+        for filter_row in self.filters["multi_choice"][0]:
```

Comment:
I'm having a hard time understanding the structure of this filter and how many rows and columns there are inside. Let's have a call later and draw it together to clarify the logic :) I feel like we don't need to use loops here.

Reply:
Sure thing! So `self.filters["multi_choice"]` holds an Accordion component now which is basically a set. Here is a log of it `logging.info(f"multi_choice: {type({self.filters['multi_choice']})}, {self.filters['multi_choice']}")`
<img width="1049" alt="image" src="https://github.com/technologiestiftung/innovationserhebung/assets/50147356/558ed007-d2b1-47e2-82cf-7dc2148051b9">

so by looping over index 0, I iterate over every filter row, that consists of a div ( [0] HTML(str) ) and a Checkbox ( [1] Checkbox(css_classes=['legend-checkbox'], name='nahrung', value=True) ). 
So if the "value" of a checkbox is True, it is appended to the selected_lines: `if filter_row[1].value:
                    selected_lines.append(filter_row[1].name)`

Something equal happened to the single_choice_filters. Since we moved the complete rendering to the plotter.py file, `self.filters['single_choice']` now holds the Accordion and the actual RadioBoxGroup went one hierarchy lower:
`logging.info(f"single_choice: {self.filters['single_choice']}")` results in:
<img width="914" alt="image" src="https://github.com/technologiestiftung/innovationserhebung/assets/50147356/c3c61add-25ac-43d2-9228-b04489277f96">
so to actually read out the value of the multi_choice_filter, like "anzahl_unternehmen_insges" we have to look for `self.filters['single_choice'][0].value`

does that make it a little clearer?

Reply:
One more info. Before, we used panels CheckBoxGroup-Widget to render the multi_choice_filters. In order to transform it to a legend like this: 
<img width="415" alt="image" src="https://github.com/technologiestiftung/innovationserhebung/assets/50147356/791ee5e0-cb5d-4638-830c-d519dc87167f">
we need access every Row in this "Column" : 
<img width="916" alt="image" src="https://github.com/technologiestiftung/innovationserhebung/assets/50147356/d1c4cc27-7e68-4146-b004-4304123bff51">
This is the cause of that additional nesting